src/heuristicblocking.js

```diff
@@ -18,64 +18,182 @@ require.scopes.heuristicblocking = (function() {
 
 /*********************** heuristicblocking scope **/
 
-var Utils = require("utils").Utils;
-var pbStorage = require("storage");
+// make hueristic obj with utils and storage properties and put the things on it
 var webrequest = require("webrequest");
+var utils = require("utils");
 var tabOrigins = { }; // TODO roll into tabData?
 
-/**
- * Adds Cookie blocking for all more specific domains than the blocked origin
- * - if they're on the cb list
- *
- * @param {String} origin Origin to check
- */
-function setupSubdomainsForCookieblock(origin){
-  var cbl = pbStorage.getBadgerStorageObject("cookieblock_list");
-  for(var domain in cbl.getItemClones()){
-    if(origin == window.getBaseDomain(domain)){
-      pbStorage.setupHeuristicAction(domain, pb.COOKIEBLOCK);
-    }
-  }
-  // iterate through all elements of cookie block list
-  // if element has basedomain add it to action_map
-  // or update it's action with cookieblock
-  origin = null;
-  return false;
+function HeuristicBlocker(utils, pbStorage) {
+  this.utils = utils;
+  this.storage = pbStorage;
 }
 
-/**
- * Decide if to blacklist and add blacklist filters
- * @param {String} baseDomain The base domain (etld+1) to blacklist
- * @param {String} fqdn The FQDN
- */
-var blacklistOrigin = function(baseDomain, fqdn) { /* jshint ignore:line */
-  var cbl = pbStorage.getBadgerStorageObject("cookieblock_list");
+HeuristicBlocker.prototype = {
+  /**
+   * Adds Cookie blocking for all more specific domains than the blocked origin
+   * - if they're on the cb list
+   *
+   * @param {String} origin Origin to check
+   */
+  setupSubdomainsForCookieblock: function(origin) {
+    var cbl = this.storage.getBadgerStorageObject("cookieblock_list");
+    for(var domain in cbl.getItemClones()){
+      if(origin == window.getBaseDomain(domain)){
+        this.storage.setupHeuristicAction(domain, constants.COOKIEBLOCK);
+      }
+    }
+    // iterate through all elements of cookie block list
+    // if element has basedomain add it to action_map
+    // or update it's action with cookieblock
+    origin = null;
```

Comment:
I don't get it either. It is actually in *master* as well.

All I did here was put this function in the `HueristicBlocker` prototype, so it could access some state associated with its badger instance.